api/v1/databaseengine_types.go

```diff
@@ -73,6 +75,23 @@ type DatabaseEngineList struct {
 	Items           []DatabaseEngine `json:"items"`
 }
 
+// Versions struct represents available versions of database engine components.
+type Versions struct {
+	Engine map[string]*Component            `json:"engine,omitempty"`
+	Backup map[string]*Component            `json:"backup,omitempty"`
+	Proxy  map[string]map[string]*Component `json:"proxy,omitempty"`
+	Tools  map[string]map[string]*Component `json:"tools,omitempty"`
+}
+
+// Component contains information of the database engine component.
+// Database Engine component can be database engine, database proxy or tools image path
+type Component struct {
+	Critical  bool   `json:"critical,omitempty"`
+	ImageHash string `json:"imageHash,omitempty"`
+	ImagePath string `json:"imagePath,omitempty"`
+	Status    string `json:"status,omitempty"`
```

Comment:
Should we enum the states instead of just accepting random strings?

Reply:
Yeah. Good point